src/main/scala/rps/GameRepository.scala

```diff
@@ -5,19 +5,21 @@ import scala.collection.concurrent.TrieMap
 import model._
 
 trait GameRepository {
-  def save(play: Play): Unit
-  def read(): Option[Play]
+  def save(play: Play): Int
+  def read(userId: Int): Option[Play]
 }
 
 class InMemoryGameRepository extends GameRepository {
-  private val map = TrieMap.empty[String, Play]
+  private val map = TrieMap.empty[Int, Play]
+  private var nextUserId = 0
 
-  override def save(play: Play): Unit = {
-    map.put("play", play)
-    ()
+  override def save(play: Play): Int = {
+    map.put(nextUserId, play)
+    nextUserId = nextUserId + 1
```

Comment:
Probably will be better to have a specific case class for  game id, take  also a look to AtomicInteger for id generation

Reply:
`case UserId(userId: Int)` seems overkill though. Is that what you mean?